library/src/uk/co/senab/bitmapcache/BitmapLruCache.java

```diff
@@ -521,29 +521,6 @@ synchronized void setDiskCache(DiskLruCache diskCache) {
             mDiskCacheEditLocks = new HashMap<String, ReentrantLock>();
             mDiskCacheFlusherExecutor = new ScheduledThreadPoolExecutor(1);
             mDiskCacheFlusherRunnable = new DiskCacheFlushRunnable(diskCache);
-
-            String cacheDirParent = mDiskCache.getDirectory().getParent();
-            final String cacheDirName = mDiskCache.getDirectory().getName();
-            mFileObserver = new FileObserver(cacheDirParent, FileObserver.DELETE) {
-                @Override
-                public void onEvent(int event, String path) {
-                    //Log.d("Test", String.format("FileObserver event 0x%08x %s", event, path));
-                    if (cacheDirName.equals(path) && ((FileObserver.DELETE & event) != 0)) {
-                        //Log.d("Test", "Deleted " + cacheDirName);
-                        try {
-                            mDiskCache.close();
-                        } catch (IOException e) {
-                            //Log.e("Test", "Error", e);
-                        }
-                        mDiskCache = null;
-                        mDiskCacheEditLocks = null;
-                        mDiskCacheFlusherExecutor = null;
-                        mDiskCacheFlusherRunnable = null;
-                        mFileObserver.stopWatching();
-                    }
-                }
-            };
-            mFileObserver.startWatching();
```

Comment:
I tried clearing the cache in the sample app to try to illicit some kind of issue, but I did not observe anything.